File: src/lessons/welcome/methods/args/MethodsArgsEntity.py
def run():
    global forward, backward
    _forward  = forward
    _backward = backward

    def forward(i=-1):
        if i!=-1:
            errorMsg("forward(int) forbidden in this exercise")
        else:
            _forward()

    def backward(i=-1):
        if i!=-1:
            errorMsg("backward(int) forbidden in this exercise")
        else:
            _backward()

    # BEGIN SOLUTION 
    def move(nbPas, doforward):
        if doforward:
            for i in range(nbPas):
                forward()
        else:
            for i in range(nbPas):
                backward()
    # END SOLUTION

    move(getY(), getDirection() == Direction.NORTH) 

File: src/lessons/welcome/methods/args/MethodsArgsEntity.c
#include "../../../../../target/classes/resources/langages/c/RemoteBuggle.h"

/* BEGIN TEMPLATE */
/* BEGIN SOLUTION */
void move(int nbStep, int f)
{

  int i;
  if (f) {
    for (i = 0; i < nbStep; i++)
      stepForward();
  } else {
    for (i = 0; i < nbStep; i++)
      stepBackward();
  }
}
/* END SOLUTION */
/* END TEMPLATE */

void run()
{
  if (getDirection() == NORTH) {
    move(getY(), 1);
  } else {
    move(getY(), 0);
  }
}
